src/linker/Linker.Steps/SweepStep.cs

```diff
@@ -925,7 +927,10 @@ void UpdateScopeOfTypeReference (TypeReference type)
 				//
 				// Resolve to type definition to remove any type forwarding imports
 				//
-				TypeDefinition td = resolver.TryResolve (type);
+				// Workaround for https://github.com/mono/linker/issues/2260
```

Comment:
nit: it would be helpful to add a short comment about why the cache shouldn't be used here - something like "This needs to use Cecil's Resolve because the LinkContext.Resolve cache has been invalidated by sweeping unused types".